packages/fx-core/src/plugins/resource/sql/v3/index.ts

```diff
@@ -0,0 +1,434 @@
+  ): Promise<Result<v2.ResourceTemplate[], FxError>> {
+    const solutionSettings = ctx.projectSetting.solutionSettings as
+      | AzureSolutionSettings
+      | undefined;
+    const pluginCtx = { plugins: solutionSettings ? solutionSettings.activeResourcePlugins : [] };
+    const bicepTemplateDirectory = path.join(
+      getTemplatesFolder(),
+      "plugins",
+      "resource",
+      "sql",
+      "bicep"
+    );
+    const provisionOrchestration = await generateBicepFromFile(
+      path.join(bicepTemplateDirectory, AzureSqlBicepFile.moduleTemplateFileName),
+      pluginCtx
+    );
+    const provisionModules = await generateBicepFromFile(
+      path.join(bicepTemplateDirectory, AzureSqlBicepFile.ProvisionModuleTemplateFileName),
+      pluginCtx
+    );
+    const result: ArmTemplateResult = {
+      Provision: {
+        Orchestration: provisionOrchestration,
+        Modules: { azureSql: provisionModules },
+      },
+      Parameters: await fs.readJSON(path.join(bicepTemplateDirectory, Bicep.ParameterFileName)),
+      Reference: {
+        sqlResourceId: AzureSqlBicep.sqlResourceId,
+        sqlEndpoint: AzureSqlBicep.sqlEndpoint,
+        databaseName: AzureSqlBicep.databaseName,
+      },
+    };
+    return ok([{ kind: "bicep", template: result }]);
+  }
+
+  public async generateNewDatabaseBicep(
+    ctx: v3.ContextWithManifestProvider
+  ): Promise<Result<v2.ResourceTemplate[], FxError>> {
+    const suffix = getUuid().substring(0, 6);
+    const compileCtx = {
+      suffix: suffix,
+    };
+    const bicepTemplateDirectory = path.join(
+      getTemplatesFolder(),
+      "plugins",
+      "resource",
+      "sql",
+      "bicep"
+    );
+    const provisionOrchestration = await generateBicepFromFile(
+      path.join(bicepTemplateDirectory, AzureSqlBicepFile.newDatabaseOrchestrationTemplateFileName),
+      compileCtx
+    );
+    const provisionModules = await generateBicepFromFile(
+      path.join(bicepTemplateDirectory, AzureSqlBicepFile.newDatabaseProvisionTemplateFileName),
+      compileCtx
+    );
+    const result: ArmTemplateResult = {
+      Provision: {
+        Orchestration: provisionOrchestration,
+        Modules: { azureSql: provisionModules },
+      },
+      Reference: {
+        sqlResourceId: AzureSqlBicep.sqlResourceId,
+        sqlEndpoint: AzureSqlBicep.sqlEndpoint,
+        databaseName: AzureSqlBicep.databaseName,
+      },
+    };
+    return ok([{ kind: "bicep", template: result }]);
+  }
+  @hooks([CommonErrorHandlerMW({ telemetry: { component: BuiltInFeaturePluginNames.sql } })])
+  async addFeature(
+    ctx: v3.ContextWithManifestProvider,
+    inputs: v2.InputsWithProjectPath
+  ): Promise<Result<v2.ResourceTemplate[], FxError>> {
+    const solutionSettings = ctx.projectSetting.solutionSettings as AzureSolutionSettings;
+    const activeResourcePlugins = solutionSettings.activeResourcePlugins;
+    const firstTime = activeResourcePlugins.includes(this.name);
```

Comment:
Currently, we use existingResource in ctx.answer because the activeResourcePlugins contain SQL if SQL resource is selected.  
Here we use the activeResourcePlugins directly. The meaning of activeResourcePlugins has changed?